internal/gitutil/git_test.go

```diff
@@ -0,0 +1,464 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitutil
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+const (
+	testRemoteRepo  = "./testdata/test-remote-repo"
+	testClonedRepo  = "./testdata/test-cloned-repo"
+	testUnknownRepo = "./testdata/unknown-repo"
+	testEmptyRepo   = "./testdata/test-folder"
+	testNewFile     = "example-git-file"
+)
+
+// setupGitRepo is a helper function used to setup the test git repo
+func setupRemoteRepos(t *testing.T) {
+	t.Helper()
+	// get the pwd
+	pwd := os.Getenv("PWD")
```

Comment:
why do we need current directory here? can't we do

```go
func tempDir(t *testing.T) string{
  t.Helper()
  dir := ioutil.TempDir("","git-test")
  t.Cleanup(func(){os.RemoveAll(dir)})
  return dir
}
```

Reply:
To be honest, this was my initial approach. But i found that git had some trouble understanding the relative path when the remote is configured.

Reply:
Addressed with the suggested changes.